youtube_dl/options.py

```diff
@@ -350,6 +350,10 @@ def _scrub_eq(o):
         '--video-password',
         dest='videopassword', metavar='PASSWORD',
         help='Video password (vimeo, smotri, youku)')
+    authentication.add_option(
+        '--auth-provider',
+        dest='auth_provider', metavar='AUTH_PROVIDER',
+        help='Authentication provider (nhl, rogers, ...)')
```

Comment:
`--ap-mso` should not be touched since it's a separate stand-alone mechanism that has stable unique MSO identifiers for TV providers across all extractors while this options is not.